scripts/xcode_cloud_helper.rb

```diff
@@ -0,0 +1,763 @@
+            end
+
+            name = workflow["attributes"]["name"]
+            unless local_workflows.include? name
+                puts "Deleting unused workflow #{workflow["id"]} #{name}"
+                delete_workflow(workflow["id"])
+            end
+        }
+    else
+        puts "No"
+    end
+end
+
+def get_action_for_target(name)
+    workflow_id = get_workflow_id_for_name(name)
+    if workflow_id.nil?
+        get_new_action_for_target(name)
+    else
+        workflow_info = get_workflow_info(workflow_id)
+        result = JSON.parse(workflow_info)
+        build_action = result["data"]["attributes"]["actions"]
+        return build_action
+    end
+end 
+
+def get_new_action_for_target(name)
+    target_split = name.split('-')
+    platform = target_split[0]
+    target = target_split[1]
+
+    name = ''
+    build_platform = ''
+    test_destination = ''
+    case platform
+    when 'osx'
+        name = 'Test - macOS'
+        build_platform = 'MACOS'
+        test_destination = {
+            "deviceTypeName" => "Mac",
+            "deviceTypeIdentifier" => "mac",
+            "runtimeName" => "Same As Selected macOS Version",
+            "runtimeIdentifier" => "builder",
+            "kind" => "MAC"
+        }
+    when 'catalyst'
+        name = 'Test - macOS (Catalyst)'
+        build_platform = 'MACOS'
+        test_destination = {
+            "deviceTypeName" => "Mac (Mac Catalyst)",
+            "deviceTypeIdentifier" => "mac_catalyst",
+            "runtimeName" => "Same As Selected macOS Version",
+            "runtimeIdentifier" => "builder",
+            "kind" => "MAC"
+        }
+    when 'ios'
+        name = 'Test - iOS'
+        build_platform = 'IOS'
+        test_destination = {
+            "deviceTypeName" => "iPhone 11",
+            "deviceTypeIdentifier" => "com.apple.CoreSimulator.SimDeviceType.iPhone-11",
+            "runtimeName" => "Latest from Selected Xcode (iOS 16.1)",
+            "runtimeIdentifier" => "default",
+            "kind" => "SIMULATOR"
+        }
+    when 'tvos'
+        name = 'Test - tvOS'
+        build_platform = 'TVOS'
+        test_destination = {
+            "deviceTypeName" => "Recommended Apple TVs",
+            "deviceTypeIdentifier" =>  "recommended_apple_tvs",
+            "runtimeName" =>  "Latest from Selected Xcode (tvOS 16.4)",
+            "runtimeIdentifier" =>  "default",
+            "kind" =>  "SIMULATOR"
+        }
+    else #docs, swiftlint, cocoapods, swiftpm, spm, xcframework, objectserver, watchos
+        return [{
+            "name" =>  "Build - macOS",
+            "actionType" => "BUILD",
+            "destination" => "ANY_MAC",
+            "buildDistributionAudience" => nil,
+            "testConfiguration" => nil,
+            "scheme" => "CI",
+            "platform" => "MACOS",
+            "isRequiredToPass" => true
+        }]
+    end
+
+    scheme = ''
+    case target
+    when 'swift'
+        scheme = "RealmSwift"
```

Comment:
Ruby switches are expressions so this can be
```suggestion
    scheme = case target
    when 'swift'
        "RealmSwift"
```